app/controlplane/internal/server/grpc.go

```diff
@@ -192,16 +194,52 @@ func craftMiddleware(opts *Opts) []middleware.Middleware {
 	middlewares = append(middlewares,
 		// if we require a robot account
 		selector.Server(
-			// 1 - Extract the robot account from the JWT
-			jwtMiddleware.Server(func(_ *jwt.Token) (interface{}, error) {
-				// TODO: add support to multiple signing methods and keys
-				return []byte(opts.AuthConfig.GeneratedJwsHmacSecret), nil
-			},
-				jwtMiddleware.WithSigningMethod(robotaccount.SigningMethod),
-				jwtMiddleware.WithClaims(func() jwt.Claims { return &robotaccount.CustomClaims{} }),
+			// 1 - Extract information from the JWT by using the claims
+			usercontext.WithJWTTokenMiddleware(
+				// Robot account provider
+				usercontext.WithTokenProvider(
```

Comment:
Probably you can move this (and the one velow) to a function for better clarity.